File: simulator/ts/plan/planner-straight.ts
import { PointV } from "../dtypes"
import { Path } from "../geo/path"
import { SegmentLine } from "../geo/segment-line"

/**
 * Fly straight forever.
 */
export function straight(loc: PointV): Path {
  // Project velocity
  const dest = {
    x: loc.x + loc.vx,
    y: loc.y + loc.vy
  }
  return new Path("Str", new SegmentLine(loc, dest))
}

File: android/app/src/main/java/ws/baseline/autopilot/plan/PlannerStraight.java
package ws.baseline.autopilot.plan;

import ws.baseline.autopilot.geo.Path;
import ws.baseline.autopilot.geo.Point;
import ws.baseline.autopilot.geo.PointV;
import ws.baseline.autopilot.geo.SegmentLine;

import androidx.annotation.NonNull;

class PlannerStraight {

    /**
     * Fly straight forever.
     */
    @NonNull
    static Path straight(PointV loc) {
        final Point target = new Point(loc.x + loc.vx, loc.y + loc.vy);
        return new Path(
                new SegmentLine(loc, target)
        );
    }
}
